go/integTest/pubsub_basic_test.go

```diff
@@ -3,290 +3,964 @@
+			readMethod:     WaitForMessageMethod,
+			useCallback:    false,
+			channelName:    "test-pattern-channel",
+			messageContent: "test pattern message",
+		},
+		{
+			name:           "Cluster with SignalChannel",
+			clientType:     GlideClusterClient,
+			readMethod:     SignalChannelMethod,
+			useCallback:    false,
+			channelName:    "test-pattern-channel",
+			messageContent: "test pattern message",
+		},
+		{
+			name:           "Cluster with SyncLoop",
+			clientType:     GlideClusterClient,
+			readMethod:     SyncLoopMethod,
+			useCallback:    false,
+			channelName:    "test-pattern-channel",
+			messageContent: "test pattern message",
+		},
 	}
 
-	callback := func(message *api.PubSubMessage, ctx any) {
-		messageReceived <- message
-	}
+	for _, tt := range tests {
+		suite.T().Run(tt.name, func(t *testing.T) {
+			publisher := suite.createAnyClient(tt.clientType, nil)
 
-	subscriptionConfig := api.NewStandaloneSubscriptionConfig().
-		WithSubscription(api.ExactChannelMode, "test-channel").
-		WithCallback(callback, &messageContext)
+			channels := []ChannelDefn{
+				{Channel: tt.channelName, Mode: ExactMode},
+			}
+			expectedMessages := map[string]string{
+				tt.channelName: tt.messageContent,
+			}
 
-	suite.createClientWithSubscriptions("subscriber", subscriptionConfig)
+			var receiver api.BaseClient
+			queues := make(map[int]*api.PubSubMessageQueue)
+			if !tt.useCallback {
+				receiver = suite.CreatePubSubReceiver(tt.clientType, channels, 1, false)
+				queue, err := receiver.GetQueue()
+				assert.Nil(t, err)
+				queues[1] = queue
+			} else {
+				receiver = suite.CreatePubSubReceiver(tt.clientType, channels, 1, true)
+			}
 
-	// Give some time for subscription to be established
-	time.Sleep(100 * time.Millisecond)
+			// Allow subscription to establish
+			time.Sleep(100 * time.Millisecond)
 
-	// Publish a message
-	testMessage := "hello world"
-	result, err := publisher.Publish("test-channel", testMessage)
-	// result, err := publisher.CustomCommand([]string{"PUBLISH", "test-channel", testMessage})
-	assert.Nil(suite.T(), err)
-	assert.NotNil(suite.T(), result)
+			// Publish test message
+			_, err := publisher.Publish(tt.channelName, tt.messageContent)
+			assert.Nil(t, err)
 
-	// Wait for the message to be received
-	var receivedMessage *api.PubSubMessage
-	select {
-	case receivedMessage = <-messageReceived:
-	case <-time.After(5 * time.Second):
-		assert.Fail(suite.T(), "Timed out waiting for message")
-	}
+			// Allow time for the message to be received
+			time.Sleep(100 * time.Millisecond)
+
+			// Verify using the verification function
+			suite.verifyPubsubMessages(t, expectedMessages, queues, tt.readMethod)
 
-	// Verify the message content
-	if receivedMessage != nil {
-		assert.Equal(suite.T(), testMessage, receivedMessage.Message)
-		assert.Equal(suite.T(), "test-channel", receivedMessage.Channel)
-		assert.True(suite.T(), receivedMessage.Pattern.IsNil())
+			// Clean up
+			if receiver != nil {
```

Comment:
Need to close publisher too

Reply:
These cleanup lines are unneeded and have been removed. All clients (publisher or receiver) are added to a slice in the test suite and closed during test teardown. No need for the extra code in the test.